### PR TITLE
Pocketxtra 27 component cards

### PR DESCRIPTION
**Pull request by tabassum2507**
@rohan-arya-w blur effect is added in the background

### CODE REVIEW

File: src/components/cards/CardComponentTemp.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  responsiveHeight,
+  responsiveWidth,
+  responsiveFontSize
+} from "react-native-responsive-dimensions";
+import { BlurView } from 'expo-blur';
+import { OrgEventInfoCardInterface } from '../../interface/cards/TempInterface';
+
+export const OrgEventInfoCard: React.FC<OrgEventInfoCardInterface> = ({
+  EventDate = "24 Jun 2023 - 28 Jun 2023",
+  EventLocation = "Delhi",
```
Comment (rohan-arya-w) [inline src/components/cards/CardComponentTemp.tsx]: We don't need hardcoded values here

File: src/components/cards/DashBoardInfoCard.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+const styles = StyleSheet.create({
+  cardContent: {
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+    padding: responsiveWidth(4),
+    height: 'auto',
+    minHeight: 'auto',
+
```
Comment (rohan-arya-w) [inline src/components/cards/DashBoardInfoCard.tsx]: Remove extra Line

File: src/components/cards/index.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-export { default as ContainedButton } from '../buttons/containedButton';
+export {OrgEventInfoCard} from './CardComponentTemp';
+export {DashBaordInfoCard} from './DashBoardInfoCard'
```
Comment (rohan-arya-w) [inline src/components/cards/index.ts]: Semi colon missing

File: src/components/cards/DashBaordUpcommingEventCard.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import {
+  responsiveHeight,
+  responsiveWidth,
+  responsiveFontSize
+} from "react-native-responsive-dimensions";
+import { BlurView } from 'expo-blur';
+import { DashBoardInfoCardInterface } from '../../interface/cards/DashBoardInfoCardInterface';
+
+export const DashBaordUpcommingEventCard: React.FC<DashBoardInfoCardInterface> =({
+  width = 50,
```
Comment (rohan-arya-w) [inline src/components/cards/DashBaordUpcommingEventCard.tsx]: remove width size

File: src/components/cards/EditAndUpCommingEventCard.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      <BlurView style={[styles.infoSection, {borderRadius: responsiveWidth(3), overflow: "hidden", flex: 1,}]} intensity={30} tint="dark">
+      
+        <Text style={{ color: color, marginRight: responsiveWidth(2), marginLeft: responsiveWidth(2),  fontFamily: fontFamily, fontSize: responsiveFontSize(fontSize) }}>{EventLocation}</Text>
+        <Text style={{ color: color, marginRight: responsiveWidth(2),  fontFamily: fontFamily , fontSize: responsiveFontSize(fontSize)}}>{Category}</Text>
+        <Text style={{ color: color, marginRight: responsiveWidth(2),  fontFamily: fontFamily, fontSize: responsiveFontSize(fontSize) }}>{Company}</Text>
+       
+      </BlurView>
+
+        {
+          EventStatus === "UpComming" ? (
```
Comment (rohan-arya-w) [inline src/components/cards/EditAndUpCommingEventCard.tsx]: add all the const in const file

File: src/index.ts [hunk 6]
```diff
@@ -1 +1 @@
-import './components/buttons/index' // import all buttons component here to the entry point for the npm package
\ No newline at end of file
+import './components/cards/index' // import all buttons component here to the entry point for the npm package
```
Comment (rohan-arya-w) [inline src/index.ts]: Please add the correct comment

File: src/interface/cards/DashBoardInfoCardInterface.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    width?: number,
+    height?: number,
+    EventName?: string,
+    color?: string,
+    EventDate?: string,
+    colorDark?: string,
+    CompanyName?: string,
+    borderRadius?: number,
+    imageSource?: ImageSource,
+    fontFamily?: string,
```
Comment (rohan-arya-w) [inline src/interface/cards/DashBoardInfoCardInterface.ts]: Do we need this ??

File: src/interface/cards/TempInterface.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+export interface OrgEventInfoCardInterface{
+    EventDate?: string,
+    EventLocation?: string,
+    Category?: string,
+    Company?: string,
+    Eventdes?: string,
+    fontFamily?: string,
```
Comment (rohan-arya-w) [inline src/interface/cards/TempInterface.ts]: Remove font family from here

File: src/components/cards/DashBaordUpcommingEventCard.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  height = 40,
+  EventName = "",
+  color= "",
+  EventDate = "",
+  CompanyName = "",
+  colorDark = '',
```
Comment (rohan-arya-w) [inline src/components/cards/DashBaordUpcommingEventCard.tsx]: give something meaning full name

Comment (rohan-arya-w) [reply src/components/cards/DashBaordUpcommingEventCard.tsx]: like textColor

File: src/components/cards/DashBaordUpcommingEventCard.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  borderRadius = 8,
+  imageSource = { uri: '' },
+  fontFamily = 'Nunito'
```
Comment (rohan-arya-w) [inline src/components/cards/DashBaordUpcommingEventCard.tsx]: Remove

File: src/components/cards/DashBaordUpcommingEventCard.tsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  cardContent: {
+    position: 'absolute',
+    bottom: 0,
+    left: 0,
+    right: 0,
+    padding: responsiveWidth(4),
+    height: 'auto',
+    minHeight: 'auto',
+   
+
```
Comment (rohan-arya-w) [inline src/components/cards/DashBaordUpcommingEventCard.tsx]: Remove extra line

File: src/components/cards/EditAndUpCommingEventCard.tsx [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+            <BlurView style={[styles.cardContent, { borderBottomLeftRadius : responsiveWidth(3),
```
Comment (rohan-arya-w) [inline src/components/cards/EditAndUpCommingEventCard.tsx]: Remove inline styles

File: src/components/cards/EditAndUpCommingEventCard.tsx [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+            <Text style={{ color: 'white', fontFamily: fontFamily, fontSize: responsiveFontSize(DesTextSize) }}>{Eventdes}</Text>
+            <View style={styles.Posted}>
+            <Text style={{ color: EventColor, marginRight: responsiveWidth(6),  fontFamily: fontFamily, fontSize: responsiveFontSize(1.4), }}>Posted By</Text>
+            <Text style={{ color: color, marginRight: responsiveWidth(6),  fontFamily: fontFamily , fontSize: responsiveFontSize(1.8)}}>{CompanyName}</Text>
+          </View>
+          </BlurView>
+          ) :  (
+            <View style={[styles.cardContent]}>
+             
+             <Text style={{ color: color, fontFamily: fontFamily, fontSize: responsiveFontSize(DesTextSize) }}>{Eventdes}</Text>
```
Comment (rohan-arya-w) [inline src/components/cards/EditAndUpCommingEventCard.tsx]: Remove inline styles

File: src/components/cards/EditAndUpCommingEventCard.tsx [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+
+    </Card>
+  );
+};
+
+const styles = StyleSheet.create({
+  dateSection: {
+    position: 'absolute',
+    top: 15,
+
```
Comment (rohan-arya-w) [inline src/components/cards/EditAndUpCommingEventCard.tsx]: remove extra line

File: src/components/cards/EditAndUpCommingEventCard.tsx [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+    bottom: 0,
+    left: 0,
+    right: 0,
+    padding: responsiveWidth(4),
+    height: 'auto',
+    minHeight: 'auto',
+  },
+  Posted:{
+    alignItems: 'flex-end',
+
```
Comment (rohan-arya-w) [inline src/components/cards/EditAndUpCommingEventCard.tsx]: Remove extra space

File: src/interface/cards/TempInterface.ts [hunk 16]
```diff
@@ -0,0 +1,24 @@
+export interface OrgEventInfoCardInterface{
+    EventDate?: string,
+    EventLocation?: string,
+    Category?: string,
+    Company?: string,
+    Eventdes?: string,
```
Comment (rohan-arya-w) [inline src/interface/cards/TempInterface.ts]: Give proper naming

**Review by rohan-arya-w — CHANGES_REQUESTED** [review]
format all files

File: src/components/cards/CardComponentTemp.tsx (hunk 1 above)
Comment (tabassum2507) [reply src/components/cards/CardComponentTemp.tsx]: hardcoded values are removed

File: src/components/cards/DashBoardInfoCard.tsx (hunk 2 above)
Comment (tabassum2507) [reply src/components/cards/DashBoardInfoCard.tsx]: Extra lines are removed from line no. 48 49

File: src/components/cards/index.ts (hunk 3 above)
Comment (tabassum2507) [reply src/components/cards/index.ts]: added

File: src/index.ts (hunk 6 above)
Comment (tabassum2507) [reply src/index.ts]: resolved

File: src/interface/cards/TempInterface.ts (hunk 8 above)
Comment (tabassum2507) [reply src/interface/cards/TempInterface.ts]: resolved

File: src/components/cards/DashBaordUpcommingEventCard.tsx
Comment (Amitrawat21) [inline src/components/cards/DashBaordUpcommingEventCard.tsx:1]: upcoming spelling mistake

Comment (Amitrawat21) [inline src/components/cards/DashBaordUpcommingEventCard.tsx:1]: dashboard spelling mistake

File: src/components/cards/DashBaordUpcommingEventCard.tsx [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+DashBaordUpcommingEventCardInterface
+> = ({
+  width = 100,
+  height = 70,
+  eventName = '',
+  textColor = Colors.textColor,
+  eventDate = '',
+  companyName = '',
+  borderRadius = 8,
+  imageSource = { uri: 'https://picsum.photos/700' },
```
Comment (ShubhamBade) [inline src/components/cards/DashBaordUpcommingEventCard.tsx]: Create assets folder and store the reuired image. Then pass image path here. Dont not pass internet url. @rohan-arya-w  Please confirm

File: src/components/cards/DashBaordUpcommingEventCard.tsx
Comment (tabassum2507) [reply src/components/cards/DashBaordUpcommingEventCard.tsx:1]: Resolved

Comment (tabassum2507) [reply src/components/cards/DashBaordUpcommingEventCard.tsx:1]: resolved

File: src/interface/cards/TempInterface.ts (hunk 16 above)
Comment (tabassum2507) [reply src/interface/cards/TempInterface.ts]: resolved

File: src/components/cards/EditAndUpCommingEventCard.tsx (hunk 15 above)
Comment (tabassum2507) [reply src/components/cards/EditAndUpCommingEventCard.tsx]: extra space removed

File: src/components/cards/EditAndUpCommingEventCard.tsx (hunk 14 above)
Comment (tabassum2507) [reply src/components/cards/EditAndUpCommingEventCard.tsx]: resolved

File: src/components/cards/DashBaordUpcommingEventCard.tsx [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+        style={[
+          styles.cardContent,
+          {
+            borderBottomLeftRadius: responsiveWidth(borderRadius),
+            borderBottomRightRadius: responsiveWidth(borderRadius),
+            overflow: "hidden",
+            flex: 1,
+          },
+        ]}
+        intensity={30}
```
Comment (ShubhamBade) [inline src/components/cards/DashBaordUpcommingEventCard.tsx]: @rohan-arya-w  do we need to change blur intensity to dynamic?

File: src/components/cards/EditAndUpCommingEventCard.tsx (hunk 5 above)
Comment (tabassum2507) [reply src/components/cards/EditAndUpCommingEventCard.tsx]: resolved

File: src/components/cards/DashBaordUpcommingEventCard.tsx (hunk 4 above)
Comment (tabassum2507) [reply src/components/cards/DashBaordUpcommingEventCard.tsx]: width is 100% now

File: src/interface/cards/DashBoardInfoCardInterface.ts (hunk 7 above)
Comment (tabassum2507) [reply src/interface/cards/DashBoardInfoCardInterface.ts]: font-family removed

File: src/components/cards/DashBaordUpcommingEventCard.tsx (hunk 9 above)
Comment (tabassum2507) [reply src/components/cards/DashBaordUpcommingEventCard.tsx]: name changed

File: src/components/cards/DashBaordUpcommingEventCard.tsx (hunk 11 above)
Comment (tabassum2507) [reply src/components/cards/DashBaordUpcommingEventCard.tsx]: resolved

File: src/components/cards/EditAndUpcomingEventCard.tsx [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+             fontSize: responsiveFontSize(eventDateFontSize)
+             }}>
+          {eventDate}
+        </Text>
+      </View>
+
+      <BlurView
+        style={[styles.infoSection,
+        {
+          borderRadius: responsiveWidth(3),
```
Comment (ShubhamBade) [inline src/components/cards/EditAndUpcomingEventCard.tsx]: make it dynamic -borderRadius

File: src/components/cards/DashBaordUpcommingEventCard.tsx (hunk 10 above)
Comment (tabassum2507) [reply src/components/cards/DashBaordUpcommingEventCard.tsx]: font family removed

File: src/components/cards/EditAndUpcomingEventCard.tsx (hunk 19 above)
Comment (tabassum2507) [reply src/components/cards/EditAndUpcomingEventCard.tsx]: Resolved

**Review by rohan-arya-w — APPROVED** [review]

**Review by emailpocketxtra — APPROVED** [review]

**Review by ShubhamBade — APPROVED** [review]
done.

**Pull request merged**